File: NeKanbanApi/NeKanban.Common/Entities/DeskUser.cs
using NeKanban.Common.Constants;

namespace NeKanban.Common.Entities;

public class DeskUser : IHasPk<int>
{
    public int Id { get; set; }
    public required int DeskId { get; set; }
    public required int UserId { get; set; }
    public required int? RoleId { get; set; }
    public required bool IsOwner { get; set; }
    public PreferenceType Preference { get; set; } = PreferenceType.Normal;
    public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();
    public virtual Role? Role { get; set; }
    public virtual ApplicationUser? User { get; set; }
    public virtual Desk? Desk { get; set; }
}

File: nekanban/src/app/models/deskUser.ts
import {User} from "./user";
import {Desk} from "./desk";
import {Role} from "./Role";
import {DeletionReason} from "../constants/deletionReason";

export interface DeskUser {
  id: number;
  user: User;
  preference: number;
  preferenceName: string;
  isOwner: boolean;
  role: Role;
  deletionReason: DeletionReason;
}
